### DISCUSSION TITLE
Is the max-size-mb specified for file_cache, effective when files are still open?

### DISCUSSION THREAD

**Discussion by arvindsh** [Azure/azure-storage-fuse#1257] [General]
In the file_cache configuration (not streaming), let's say `max-size-mb` is set to 30 GiB. Let's also say an application is writing out a single 100 GiB file in "one go" i.e. with the same file handle. In such cases, I observed that the size of the local cache grows well beyond the max-size-mb set. Is this by design?

For context, I found stream mode is too slow for this app as it essentially does a lot of smaller writes, so it ends up being a bottleneck and then holds back the overall throughput of the app. So file_cache is the only performant option, however the size of the local disk needed for caching is a concern.

**Comment by bhansalivikas** [comment]
Yes cache limit in file-cache is a soft limit. If the disk usage goes beyond the limit while file is already open we do not fail any user operation. We have received a few requests from customers to make this a hard limit and fail the write/open/truncate operations if it exceeds the given limit. We are working on that but only issue with that approach is it will be at cost of performance as for every operation we will need to check the remaining disk space.

**Reply by arvindsh** [reply]
I see. I'm in agreement with the current behavior. I don't see why you'd need to check remaining disk space, though? Did you mean "for every operation we will need to check the current disk space used by the file cache"?

**Comment by bhansalivikas** [comment]
In file-cache mode all files being operated on are kept in local-cache. There might be multiple files being read/written by applicaoitn in parallel. If we wish to monitor the disk usage and fail write operations based on available disk space then every write has to recheck what amount of disk space is left now and incoming data will break that limit or not.
